storage/txcache/txListForSender.go

```diff
@@ -184,20 +196,81 @@ func (listForSender *txListForSender) findListElementWithTx(txToFind *WrappedTra
 	txToFindHash := txToFind.TxHash
 	txToFindNonce := txToFind.Tx.GetNonce()
 
+	iterations := 0
+	defer listForSender.monitorOnFoundTransaction(&iterations, txToFind)
+
 	for element := listForSender.items.Front(); element != nil; element = element.Next() {
-		value := element.Value.(*WrappedTransaction)
+		iterations++
 
-		if bytes.Equal(value.TxHash, txToFindHash) {
-			return element
+		value := element.Value.(*WrappedTransaction)
+		nonce := value.Tx.GetNonce()
+
+		// Optimization: first check nonce equality.
+		if nonce == txToFindNonce {
+			if bytes.Equal(value.TxHash, txToFindHash) {
+				return element
+			}
+		} else if nonce > txToFindNonce {
+			// Optimization: stop search at this point, since the list is sorted by nonce.
+			break
 		}
+	}
+
+	return nil
+}
 
-		// Optimization: stop search at this point, since the list is sorted by nonce
-		if value.Tx.GetNonce() > txToFindNonce {
+// RemoveSortedTransactions removes the provided list of transactions, assuming the list is given sorted by nonce
+// QUESTION FOR REVIEW: should I refactor to have a loop-in-loop when traversing the two lists instead?
```

Comment:
As you wish, I think it is also ok as it is.

Reply:
Kept.